src/Program/Model/Camera.hs

```diff
@@ -94,6 +98,17 @@ cameraBehavior cam pointerE wheelE resizeE resetCamE buttonsB coordsB alowMoveB
                                                  | b == 4 = dragVertical opos npos c
                                                    -- fallback to horizontal dragging
                                                  | otherwise = dragHorizontal opos npos c
+    cameraPositionT :: Either SomeJSONInput SomeJSONInput -> Camera -> Camera
+    cameraPositionT e c = Camera (viewportSize c) (projMatrix c) nstate nstate
+          where
+            nstate = case anyway e of
+                SJIExtended sj ->
+                    CState { viewPoint  = fromMaybe (viewPoint defaultCameraState) $ sjCameraFocus sj
```

Comment:
From the code you submitted here, it seems that you don't change the coordinates of camera focus point. The problem is that if the whole scenario is in WGS'84, camera position should also be in the same format (otherwise it is impossible to put proper camera coordinates in JSON file, right?).
So you need to perform WGS'84 -> local coordinate system transform in this case.